src/Rings/groebner.jl

```diff
@@ -154,15 +154,32 @@ function standard_basis(I::MPolyIdeal; ordering::MonomialOrdering = default_orde
   elseif algorithm == :fglm
     _compute_groebner_basis_using_fglm(I, ordering)
   elseif algorithm == :hilbert
-    if base_ring(I) isa MPolyDecRing
-      J, target_ordering  = I, ordering
+    if all(_is_homogeneous, gens(I))
```

Comment:
Wouldn't `_find_weights(gens(I))` find this, too, returning `ones(...)` as weights? If so we could make the code more compact.